[FX] Add a bunch of docstrings

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47720 [FX] Fix users not updating when erasing a node
* **#47719 [FX] Add a bunch of docstrings**

Add more docs

Differential Revision: [D24875400](https://our.internmc.facebook.com/intern/diff/D24875400)

Commit:
[FX] Add a bunch of docstrings

[ghstack-poisoned]

Commit:
Update on "[FX] Add a bunch of docstrings"


Differential Revision: [D24875400](https://our.internmc.facebook.com/intern/diff/D24875400)

[ghstack-poisoned]

Commit:
Update on "[FX] Add a bunch of docstrings"


Add more docs

Differential Revision: [D24875400](https://our.internmc.facebook.com/intern/diff/D24875400)

[ghstack-poisoned]

zdevito reviewed (approved):
More docs are good! I think PyTorch docstrings follow the "Google" docstring format: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html . The contents in these docs are fine but really should be reformated to match that format.

Commit:
Update on "[FX] Add a bunch of docstrings"


Add more docs

Differential Revision: [D24875400](https://our.internmc.facebook.com/intern/diff/D24875400)

[ghstack-poisoned]
